subworkflows/local/differential/main.nf

```diff
@@ -73,24 +75,70 @@ workflow DIFFERENTIAL {
     // Perform differential analysis with DESeq2
     // ----------------------------------------------------
 
-    // ToDo: In order to use deseq2 the downstream processes need to be updated to process the output correctly
-    // if (params.transcript_length_matrix) { ch_transcript_lengths = Channel.of([ exp_meta, file(params.transcript_length_matrix, checkIfExists: true)]).first() } else { ch_transcript_lengths = [[],[]] }
-    // if (params.control_features) { ch_control_features = Channel.of([ exp_meta, file(params.control_features, checkIfExists: true)]).first() } else { ch_control_features = [[],[]] }
-
-    // ch_samplesheet
-    //     .join(ch_counts)
-    //     .first()
-    //     .combine(ch_tools_single.deseq2)
-    //     .set { ch_counts_deseq2 }
-
-    // DESEQ2_DIFFERENTIAL (
-    //     ch_contrasts,
-    //     ch_counts_deseq2,
-    //     ch_control_features,
-    //     ch_transcript_lengths
-    // )
-    // ch_results = ch_results
-    //     .mix(DESEQ2_DIFFERENTIAL.out.results)
+    if (params.transcript_length_matrix) { ch_transcript_lengths = Channel.of([ exp_meta, file(params.transcript_length_matrix, checkIfExists: true)]).first() } else { ch_transcript_lengths = Channel.of([[],[]]) }
+    if (params.control_features) { ch_control_features = Channel.of([ exp_meta, file(params.control_features, checkIfExists: true)]).first() } else { ch_control_features = Channel.of([[],[]]) }
+
+    ch_counts
+        .join(ch_samplesheet)
+        .combine(ch_contrasts)
+        .combine(ch_transcript_lengths)
+        .combine(ch_control_features)
+        .combine(ch_tools_single.deseq2)
+        .multiMap {
+            meta_data, counts, samplesheet, meta_contrast, contrast_variable, reference, target, meta_lengths, lengths, meta_control, control, pathway, meta_tools ->
+                def meta = meta_data.clone() + meta_contrast.clone() + meta_lengths.clone() + meta_control.clone() + meta_tools.clone()
+                contrast: [ meta, contrast_variable, reference, target ]
+                samplesheet: [ meta, samplesheet, counts ]
+                control_features:   [ meta, control ]
+                transcript_lengths: [ meta, lengths ]
+                pathway: [ meta, pathway ]
+        }
+        .set { ch_deseq2 }
+
+    // do we need this process DESEQ2_NORM?
```

Comment:
Yes, we do 🙂

Reply:
why do we need both deseq2_norm and deseq2_differential? Isn't deseq2_differential also outputing the normalized data?